lib/Compiler.php

```diff
@@ -0,0 +1,195 @@
+        if ($process->run() != 0) {
+            throw new \RuntimeException('Can\'t run git log. You must ensure to run compile from xsd2php git repository clone and that git binary is available.');
+        }
+
+        $date = new \DateTime(trim($process->getOutput()));
+        $date->setTimezone(new \DateTimeZone('UTC'));
+        $this->versionDate = $date->format('Y-m-d H:i:s');
+
+        $process = new Process('git describe --tags --exact-match HEAD');
+        if ($process->run() == 0) {
+            $this->version = trim($process->getOutput());
+        }
+
+        $phar = new \Phar($pharFile, 0, 'xsd2php.phar');
+        $phar->setSignatureAlgorithm(\Phar::SHA1);
+
+        $phar->startBuffering();
+
+        $finder = new Finder();
+        $finder->files()
+            ->ignoreVCS(true)
+            ->name('*.php')
+            ->notName('Compiler.php')
+            ->in(__DIR__)
+        ;
+
+        foreach ($finder as $file) {
+            $this->addFile($phar, $file);
+        }
+
+        foreach ($finder as $file) {
+            $this->addFile($phar, $file, false);
+        }
+
+        $finder = new Finder();
+        $finder->files()
+            ->ignoreVCS(true)
+            ->name('*.php')
+            ->name('LICENSE')
+            ->exclude('Tests')
+            ->exclude('tests')
+            ->in(__DIR__.'/../vendor/composer/')
+            ->in(__DIR__.'/../vendor/doctrine/annotations')
+            ->in(__DIR__.'/../vendor/doctrine/inflector')
+            ->in(__DIR__.'/../vendor/doctrine/lexer')
+            ->in(__DIR__.'/../vendor/goetas/')
+            ->in(__DIR__.'/../vendor/jms/')
+            ->in(__DIR__.'/../vendor/phpcollection/')
+            ->in(__DIR__.'/../vendor/phpoption/')
+            ->in(__DIR__.'/../vendor/symfony/')
+            ->in(__DIR__.'/../vendor/zendframework/')
+        ;
+
+        foreach ($finder as $file) {
+            $this->addFile($phar, $file);
+        }
+
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/autoload.php'));
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/autoload_namespaces.php'));
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/autoload_psr4.php'));
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/autoload_classmap.php'));
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/autoload_real.php'));
+        if (file_exists(__DIR__.'/../vendor/composer/include_paths.php')) {
+            $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/include_paths.php'));
+        }
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/ClassLoader.php'));
+        $this->addBin($phar);
+
+        // Stubs
+        $phar->setStub($this->getStub());
+
+        $phar->stopBuffering();
+
+        // disabled for interoperability with systems without gzip ext
+        // $phar->compressFiles(\Phar::GZ);
```

Comment:
This should IMO be enabled as gzip should be loaded everywhere nowadays but this is just a personal opinion.